extensions/amp-analytics/0.1/resource-timing.js

```diff
@@ -244,15 +270,31 @@ function serialize(entries, resourceTimingSpec, win) {
 
 /**
  * Serializes resource timing entries according to the resource timing spec.
- * @param {!JsonObject} resourceTimingSpec
  * @param {!Window} win
+ * @param {!JsonObject} resourceTimingSpec
  * @return {!Promise<string>}
  */
-export function serializeResourceTiming(resourceTimingSpec, win) {
-  if (!validateResourceTimingSpec(resourceTimingSpec)) {
+export function serializeResourceTiming(win, resourceTimingSpec) {
+  if (resourceTimingSpec['done'] ||
+      !validateResourceTimingSpec(resourceTimingSpec)) {
+    resourceTimingSpec['done'] = true;
     return Promise.resolve('');
   }
-  const entries = getResourceTimingEntries(win);
+  let entries = getResourceTimingEntries(win);
+  if (entries.length >= RESOURCE_TIMING_BUFFER_SIZE) {
+    // We've exceeded the maximum buffer size so no additional metrics will be
+    // reported for this resourceTimingSpec.
+    resourceTimingSpec['done'] = true;
+  }
+
+  const responseAfter = resourceTimingSpec['responseAfter'] || 0;
+  // Update responseAfter for next time to avoid reporting the same resource
+  // multiple times.
+  resourceTimingSpec['responseAfter'] =
+      Math.max(responseAfter, performanceTimestamp(win));
```

Comment:
I have that same question that I asked before: what happen if `window.performance.now()` is not supported. I am assuming that `resourceTimingSpec['responseAfter']` value will never increase. wouldn't that cause some entries to be sent multiple times?

Reply:
I see... That's difficult to tell though...
Can we remove `performanceTimestamp` function. And check for `window.performance.now()` here. If it is not supported, add  `resourceTimingSpec['done'] = true;` after first reporting. 
And add comments as well please : )